phac-nml/irida-next
# Feature: Leave Group or Project

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds functionality for a member to be able to leave a group or project on their own without needing an owner or maintainer to remove them.

Resolves #95 and #96 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Logged in as user that is removing themselves from the group:**

![image](https://github.com/phac-nml/irida-next/assets/25466211/6d8838b8-05de-4047-a072-9481caba2925)

![remove-confirmation](https://github.com/phac-nml/irida-next/assets/25466211/fc67e6b9-dd5c-4c01-a2d6-18cf56b2c56a)


**Removing a member that is not the current logged in user**

![image](https://github.com/phac-nml/irida-next/assets/25466211/146c7317-74bb-40c7-9dc3-d44f181d58dc)

![remove-confirmation-remove-member](https://github.com/phac-nml/irida-next/assets/25466211/3bb6369f-7ac6-47a0-bc39-c0a01c37b565)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@example.com`
2. Go the the `Bacillus` group and view it's members
3. Click the ellipsis for the `admin@example.com` user and you should see a link `Leave group`
4. In the confirmation you should see `Are you sure you want to remove yourself from the group?`
5. Try step 3 for another user in the listing and you should see a link `Remove member`
6. In the confirmation you should see `Are you sure you want to remove this member from the group?`
7. Try out the same steps but for a project. Projects from the seed data only have user6 through user8 as members so you will need to log in as one of those users.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
